marray/__init__.py

```diff
@@ -79,20 +79,51 @@ def __setitem__(self, key, other):
             self.mask[key] = getattr(other, 'mask', False)
             return self.data.__setitem__(key, getattr(other, 'data', other))
 
+        def _mask_integer(self, fun):
+            data = xp.asarray(self.data, dtype=xp.int64, copy=True)
+            min = xp.min(self.data)
+            sentinel = min - 1
+            assert not xp.any(data == sentinel)
+            data[self.mask] = sentinel
+            mask = "-" * len(str(sentinel))
+            temp = fun(data).replace(str(sentinel), mask)
+            if self.dtype != xp.int64:
+                temp = temp[:-1] + f", dtype={self.dtype})"
+            return temp
+
+        def _mask_bool(self, fun):
+            data = xp.asarray(self.data, dtype=xp.uint16, copy=True)
+            data[data == 0] = 11111
+            data[data == 1] = 22222
+            data[self.mask] = 33333
+            temp = fun(data).replace("11111", "False")
+            temp = temp.replace("22222", " True")
+            temp = temp.replace("33333", "-----")
+            return temp.replace("uint16", str(self.dtype))
+
+        def _mask_float(self, fun):
+            data = xp.asarray(self.data, copy=True)
+            data[data == 0] = 0
+            data[self.mask] = float("-0")
+            pattern = re.compile(r'-0(?:\.0+)?(?:[eE][+-]?\d+)?\.?')
+            return pattern.sub('---', fun(data))
```

Comment:
Should make sure the replacement takes up the same number of characters as the original.